draft-yoshino-wish-00.txt

```diff
@@ -190,12 +190,65 @@ Internet-DrafWiSH: A General Purpose Message Framing over B October 2016
 4.  WiSH Protocol
 
    WiSH frames messages over an HTTP request or response body using the
-   framing defined in Section 5.
+   framing defined in Section 7.
 
    The Content-Type header value of the underlying HTTP request/response
-   message MUST be "application/webstream".
+   message MUST be "application/web-stream".
+
+5.  Subprotocol Negotiation
+
+   When layered over HTTP, WiSH optionally negotiates the subprotocol,
+   i.e. the protocol layered over WiSH, by using the Accept header and
+   Content-Type header of the underlying HTTP.  RFC 6455 negotiated the
+   subprotocol by exchanging the Sec-WebSocket-Protocol header.  WiSH
+   lists the offered subprotocols in the Accept header as follows:
+
+       Accept: application/web-stream;protocol=foo;q=1,
+           application/web-stream;protocol=bar;q=0.5
+
+   Following the "application/web-stream" media type, a parameter named
+   "protocol" can be specified with the subprotocol name as its value.
+   Multiple subprotocols are offered by listing multiple application/
+   web-stream media type with the protocol parameter with different
+   values.  The server chooses one subprotcol from the offered ones and
+   notifies the chosen subprotocol by including the Content-Type header
+   with the WiSH media type with the subprotocol as its value as
+   follows:
+
+       Content-Type: application/web-stream;protocol=foo
+
+
+
+
+
+Yoshino & Zhu              Expires May 1, 2017                  [Page 4]
+
+Internet-DrafWiSH: A General Purpose Message Framing over B October 2016
+
+
+   Another possible way to implement the subprotocol negotiation is to
+   specify "+web-stream" as a standard media type suffix [RFC6839].  For
+   example, "application/foo+web-stream" means that WiSH is in use and
+   the subprotocol is "foo".
+
```

Comment:
Dropped.

> foo is not a media-type, although registered as WS sub-protocol.

Did you mean that the "foo" part cannot represent a media type? It would lead to incompatibility with WebSocket if we allow a complete media type to be put in the protocol parameter. A WebSocket subprotocol is specified to be a `token`. Media types are not token but `token "/" token`.